### ISSUE TITLE
Xcode 6.3/6.4 System AlertView iOS7

### ISSUE THREAD

**Issue by RichardStoff** [calabash/calabash-ios#798]
**environment:**
* OSX: Yosemite 10.10.3 (14D136)
* Xcode path: /Applications/Xcode.app/Contents/Developer
* Xcode: 6.3.2 (6D2105)
* Calabash: 0.14.3
* Ruby: 2.1.1
* iOS simulator : 7.1
* terminal using oh my zsh

server
```
{
                    "app_version" => "7.0.6",
                        "outcome" => "SUCCESS",
                            "git" => {
        "remote_origin" => "git@github.com:calabash/calabash-ios-server.git",
               "branch" => "master",
             "revision" => "c2d009b"
    },
                         "app_id" => "com.myApp.www",
               "simulator_device" => "iPhone",
                   "app_base_sdk" => "iphonesimulator8.3",
                        "version" => "0.14.3",
                    "server_port" => 37265,
                       "app_name" => "myApp",
    "iphone_app_emulated_on_ipad" => false,
                    "form_factor" => "iphone 3.5in",
                          "4inch" => false,
           "short_version_string" => "7.0",
              "screen_dimensions" => {
         "scale" => 2,
         "width" => 640,
        "sample" => 1,
        "height" => 960
    },
                    "iOS_version" => "7.1",
                         "system" => "x86_64",
                      "simulator" => "CoreSimulator 117.15 - Device: iPhone 4s - Runtime: iOS 7.1 (11D167) - DeviceType: iPhone 4s"
}
```
(same issue with iphonesimulator8.4)

I know that AlertView on iOS8 simulator works automaticaly [issue/run_loop#66] (https://github.com/calabash/run_loop/issues/66), but doesn't seems to work on iOS7 (alertview shows up and stay that way). uia_query freeze when a AlertView system is up in console mode. 

```
irb(main):003:0> uia_query
RuntimeError: uia action failed because: Timed out running command uia.query('[]')

	from /Users/StoFF/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/calabash-cucumber-0.14.3/lib/calabash-cucumber/uia.rb:37:in `uia'
	from /Users/StoFF/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/calabash-cucumber-0.14.3/lib/calabash-cucumber/uia.rb:454:in `uia_handle_command'
	from /Users/StoFF/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/calabash-cucumber-0.14.3/lib/calabash-cucumber/uia.rb:115:in `uia_query'
	from (irb):3
	from /Users/StoFF/.rbenv/versions/2.1.1/bin/irb:11:in `<main>'
irb(main):004:0> 
```

**Comment by jmoody** [comment]
Have you seen this wiki page: [Managing Privacy Alerts: Location Services, APNS, Contacts](https://github.com/calabash/calabash-ios/wiki/Managing-Privacy-Alerts%3A--Location-Services%2C-APNS%2C-Contacts)?

What kind of privacy alert is this?  Contacts? APNS? Location?  Microphone? etc.

**Comment by jmoody** [comment]
+1 for good reports.  Thanks for using formatting and providing all the relevant information.

**Comment by RichardStoff** [comment]
(thx for the answer I try to make my report as clean as possible ^^). 

In my case I use the delay as suggested for the Geoloc part, It works fine for iOS8 at the beginning of my application launch. But for iOS7 I must accept it manualy in order to get the console prompt, otherwise the application loops. 

I also use the system alertView for Contacts, in this case I ask for contact later in the application (at this time I have the console in hand). When my Contact System AlertView show up, it stays up, then I try to check for uia_query and it freeze and time out. If I check my uia_query with no System AlertView it works fine.

```
irb(main):003:0> uia_query
[
    [0] {
             "name" => ":nil",
             "rect" => {
                 "y" => 0,
             "width" => 320,
                 "x" => 0,
            "height" => 480
        },
        "hit-point" => {
            "x" => 160,
            "y" => 240
        },
               "el" => "[object UIAWindow]",
            "label" => ":nil"
    }
]
irb(main):004:0> 
```

**Comment by jmoody** [comment]
re: Location Alert in iOS 7 
* Can you increase your delay to something absurdly long, like 10 seconds? 
* Can you paste a screenshot of the alert into this issue?

re: Contacts - a screenshot would be helpful.  Dismissing this view should be supported.

What language are the alerts?  We have spotty support for languages other than English.

**Comment by RichardStoff** [comment]
re, 
I tried to delay my alert in iOS7 in a long way, and then, I saw that my delay wasn't used at all, after a few investigations, I succeeded to add my previous normal delay and everything goes fine Thanks !
(It was a tricky one but the fact is that it was my mistake first).

For the contact Alert, we used french language, even if I changed my language it doesn't seems to change anything. BUT, correcting my Geoloc first AlertView correct also all my other system alertView, I have no idea why (maybe giving a time to UIAutomation), but now all my system alertView are behaving the same on iOS8 and iOS7.

thanks for your time, your answers lead me to the right solution ;)
